test/common/validation.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"strings"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// The interval and timeout used for polling pod logs.
+	interval = 1 * time.Second
+	timeout  = 2 * time.Minute
+)
+
+// CheckLogContent waits until logs for the logger Pod include the given content.
+// If the content is not present within timeout it returns error.
+func (client *Client) CheckLogContent(podName, content string) error {
```

Comment:
Why not have this forward to CheckLogContents, or use the varargs `...string` type to allow for a single function?

Reply:
Done with defining a couple of `checker` functions and add it as a parameter in `CheckLog` function.